Spurious clippy CI failure

In https://github.com/rust-lang/rust/pull/113569, clippy CI [failed](https://github.com/rust-lang/rust/actions/runs/5521692418/jobs/10070176140), and I don't think that can be caused by my PR:

```
FAILURES:
    tests/ui/empty_line_after_doc_comments.rs
    tests/ui/empty_line_after_outer_attribute.rs
    tests/ui/needless_arbitrary_self_type_unfixable.rs
```

<details>

```
tests/ui/empty_line_after_doc_comments.rs FAILED:
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/clippy-driver" "--error-format=json" "--emit=metadata" "-Aunused" "-Zui-testing" "-Dwarnings" "--extern" "clippy_lints=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_lints-73a44cb4c7cea627.rlib" "--extern" "clippy_lints=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_lints-73a44cb4c7cea627.rmeta" "--extern" "clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_utils-7e3201eedda9725b.rlib" "--extern" "clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_utils-7e3201eedda9725b.rmeta" "--extern" "derive_new=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libderive_new-c02f7361dc6f59be.so" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libfutures-3b3f678a2023a864.rlib" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libfutures-3b3f678a2023a864.rmeta" "--extern" "if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libif_chain-6d2bcde5417e3fc9.rlib" "--extern" "if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libif_chain-6d2bcde5417e3fc9.rmeta" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libitertools-a295c739cb0c5115.rlib" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libitertools-a295c739cb0c5115.rmeta" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-26458fd927156f14.rlib" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-26458fd927156f14.rmeta" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libquote-36054bcec486f3c0.rlib" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libquote-36054bcec486f3c0.rmeta" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libregex-4e61c4979f33639c.rlib" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libregex-4e61c4979f33639c.rmeta" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libserde-b9f428a504ba38d4.rlib" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libserde-b9f428a504ba38d4.rmeta" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libsyn-faa18c63fc12a335.rlib" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libsyn-faa18c63fc12a335.rmeta" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libtokio-ee648a3a992ea021.rlib" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libtokio-ee648a3a992ea021.rmeta" "--extern" "clippy_test_deps=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/libclippy_test_deps.rlib" "--extern" "clippy_test_deps=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_test_deps-728ed027f0623fde.rmeta" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/quote-fc00b89c3984f431" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/pulldown-cmark-5d2703cd75133d86" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/tokio-988b1ca3d2062fb8" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/serde_json-06fa4ab4a6a2c54b" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/memchr-300f0ecb3b085bd3" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/libc-467a9bf5fecaa5b9" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/serde-b5ab3846bee6426e" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/camino-95d981076bbda069" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-task-14f797a867cbbbea" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/thiserror-60234973db2aebdd" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/syn-c0e5e037aa06912e" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-core-67d40d21860bf7e2" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/proc-macro2-c56f6977d10d30ea" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-channel-18c61a4c728f3f63" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/parking_lot_core-ff12b1c9a236dc80" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/slab-084e9e34eeb9970c" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/lock_api-172e9730276f85f1" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/unicase-dc66fd2ee33a4c53" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-util-d30abe96b847bc06" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/semver-367f0564b96a3f38" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/tests/ui/empty_line_after_doc_comments" "tests/ui/auxiliary/proc_macro_attr.rs" "--edition" "2021" "--crate-type" "proc-macro" "--emit=link"



Aux build from tests/ui/empty_line_after_doc_comments.rs:1 failed
compilation of aux build failed failed with exit status: 1


full stderr:
error[E0308]: mismatched types
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:33:32
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
LL | |                 });
   | |                  ^
   | |                  |
   | |__________________expected `Span`, found `DelimSpan`
   |                    expected due to this
   |
   = note: this error originates in the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::Signature: quote::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:33:32
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
LL | |                 });
   | |                  ^
   | |                  |
   | |__________________the trait `quote::ToTokens` is not implemented for `syn::Signature`
   |                    required by a bound introduced by this call
   |
   = help: the following other types implement trait `quote::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 23 others
   = note: required for `&syn::Signature` to implement `quote::ToTokens`
   = note: this error originates in the macro `$crate::quote_token_with_context_spanned` which comes from the expansion of the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::Block: quote::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:33:32
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
LL | |                 });
   | |                  ^
   | |                  |
   | |__________________the trait `quote::ToTokens` is not implemented for `syn::Block`
   |                    required by a bound introduced by this call
   |
   = help: the following other types implement trait `quote::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 23 others
   = note: required for `&mut syn::Block` to implement `quote::ToTokens`
   = note: this error originates in the macro `$crate::quote_token_with_context_spanned` which comes from the expansion of the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `quote::__private::TokenStream: syn::__private::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:39:26
   |
LL |                 *block = parse_quote!(#my_block);
   |                          ^^^^^^^^^^^^^^^^^^^^^^^
   |                          |
   |                          the trait `syn::__private::ToTokens` is not implemented for `quote::__private::TokenStream`
   |                          required by a bound introduced by this call
   |
   = help: the following other types implement trait `syn::__private::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 298 others
   = note: this error originates in the macro `$crate::__private::quote::quote` which comes from the expansion of the macro `parse_quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::ItemTrait: quote::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:43:23
   |
LL |     TokenStream::from(quote!(#item))
   |                       ^^^^^^^^^^^^^
   |                       |
   |                       the trait `quote::ToTokens` is not implemented for `syn::ItemTrait`
   |                       required by a bound introduced by this call
   |
   = help: the following other types implement trait `quote::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 23 others
   = note: this error originates in the macro `quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::ItemImpl: quote::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:96:23
   |
LL |     TokenStream::from(quote!(#item))
   |                       ^^^^^^^^^^^^^
   |                       |
   |                       the trait `quote::ToTokens` is not implemented for `syn::ItemImpl`
   |                       required by a bound introduced by this call
   |
   = help: the following other types implement trait `quote::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 23 others
   = note: this error originates in the macro `quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 6 previous errors

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.



tests/ui/empty_line_after_outer_attribute.rs FAILED:
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/clippy-driver" "--error-format=json" "--emit=metadata" "-Aunused" "-Zui-testing" "-Dwarnings" "--extern" "clippy_lints=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_lints-73a44cb4c7cea627.rlib" "--extern" "clippy_lints=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_lints-73a44cb4c7cea627.rmeta" "--extern" "clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_utils-7e3201eedda9725b.rlib" "--extern" "clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_utils-7e3201eedda9725b.rmeta" "--extern" "derive_new=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libderive_new-c02f7361dc6f59be.so" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libfutures-3b3f678a2023a864.rlib" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libfutures-3b3f678a2023a864.rmeta" "--extern" "if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libif_chain-6d2bcde5417e3fc9.rlib" "--extern" "if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libif_chain-6d2bcde5417e3fc9.rmeta" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libitertools-a295c739cb0c5115.rlib" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libitertools-a295c739cb0c5115.rmeta" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-26458fd927156f14.rlib" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-26458fd927156f14.rmeta" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libquote-36054bcec486f3c0.rlib" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libquote-36054bcec486f3c0.rmeta" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libregex-4e61c4979f33639c.rlib" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libregex-4e61c4979f33639c.rmeta" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libserde-b9f428a504ba38d4.rlib" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libserde-b9f428a504ba38d4.rmeta" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libsyn-faa18c63fc12a335.rlib" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libsyn-faa18c63fc12a335.rmeta" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libtokio-ee648a3a992ea021.rlib" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libtokio-ee648a3a992ea021.rmeta" "--extern" "clippy_test_deps=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/libclippy_test_deps.rlib" "--extern" "clippy_test_deps=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_test_deps-728ed027f0623fde.rmeta" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/quote-fc00b89c3984f431" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/pulldown-cmark-5d2703cd75133d86" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/tokio-988b1ca3d2062fb8" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/serde_json-06fa4ab4a6a2c54b" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/memchr-300f0ecb3b085bd3" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/libc-467a9bf5fecaa5b9" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/serde-b5ab3846bee6426e" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/camino-95d981076bbda069" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-task-14f797a867cbbbea" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/thiserror-60234973db2aebdd" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/syn-c0e5e037aa06912e" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-core-67d40d21860bf7e2" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/proc-macro2-c56f6977d10d30ea" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-channel-18c61a4c728f3f63" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/parking_lot_core-ff12b1c9a236dc80" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/slab-084e9e34eeb9970c" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/lock_api-172e9730276f85f1" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/unicase-dc66fd2ee33a4c53" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-util-d30abe96b847bc06" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/semver-367f0564b96a3f38" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/tests/ui/empty_line_after_outer_attribute" "tests/ui/auxiliary/proc_macro_attr.rs" "--edition" "2021" "--crate-type" "proc-macro" "--emit=link"



Aux build from tests/ui/empty_line_after_outer_attribute.rs:1 failed
compilation of aux build failed failed with exit status: 1


full stderr:
error[E0308]: mismatched types
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:33:32
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
LL | |                 });
   | |                  ^
   | |                  |
   | |__________________expected `Span`, found `DelimSpan`
   |                    expected due to this
   |
   = note: this error originates in the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::Signature: quote::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:33:32
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
LL | |                 });
   | |                  ^
   | |                  |
   | |__________________the trait `quote::ToTokens` is not implemented for `syn::Signature`
   |                    required by a bound introduced by this call
   |
   = help: the following other types implement trait `quote::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 23 others
   = note: required for `&syn::Signature` to implement `quote::ToTokens`
   = note: this error originates in the macro `$crate::quote_token_with_context_spanned` which comes from the expansion of the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::Block: quote::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:33:32
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
LL | |                 });
   | |                  ^
   | |                  |
   | |__________________the trait `quote::ToTokens` is not implemented for `syn::Block`
   |                    required by a bound introduced by this call
   |
   = help: the following other types implement trait `quote::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 23 others
   = note: required for `&mut syn::Block` to implement `quote::ToTokens`
   = note: this error originates in the macro `$crate::quote_token_with_context_spanned` which comes from the expansion of the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `quote::__private::TokenStream: syn::__private::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:39:26
   |
LL |                 *block = parse_quote!(#my_block);
   |                          ^^^^^^^^^^^^^^^^^^^^^^^
   |                          |
   |                          the trait `syn::__private::ToTokens` is not implemented for `quote::__private::TokenStream`
   |                          required by a bound introduced by this call
   |
   = help: the following other types implement trait `syn::__private::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 298 others
   = note: this error originates in the macro `$crate::__private::quote::quote` which comes from the expansion of the macro `parse_quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::ItemTrait: quote::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:43:23
   |
LL |     TokenStream::from(quote!(#item))
   |                       ^^^^^^^^^^^^^
   |                       |
   |                       the trait `quote::ToTokens` is not implemented for `syn::ItemTrait`
   |                       required by a bound introduced by this call
   |
   = help: the following other types implement trait `quote::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 23 others
   = note: this error originates in the macro `quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::ItemImpl: quote::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:96:23
   |
LL |     TokenStream::from(quote!(#item))
   |                       ^^^^^^^^^^^^^
   |                       |
   |                       the trait `quote::ToTokens` is not implemented for `syn::ItemImpl`
   |                       required by a bound introduced by this call
   |
   = help: the following other types implement trait `quote::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 23 others
   = note: this error originates in the macro `quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 6 previous errors

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.



tests/ui/needless_arbitrary_self_type_unfixable.rs FAILED:
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/clippy-driver" "--error-format=json" "--emit=metadata" "-Aunused" "-Zui-testing" "-Dwarnings" "--extern" "clippy_lints=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_lints-73a44cb4c7cea627.rlib" "--extern" "clippy_lints=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_lints-73a44cb4c7cea627.rmeta" "--extern" "clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_utils-7e3201eedda9725b.rlib" "--extern" "clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_utils-7e3201eedda9725b.rmeta" "--extern" "derive_new=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libderive_new-c02f7361dc6f59be.so" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libfutures-3b3f678a2023a864.rlib" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libfutures-3b3f678a2023a864.rmeta" "--extern" "if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libif_chain-6d2bcde5417e3fc9.rlib" "--extern" "if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libif_chain-6d2bcde5417e3fc9.rmeta" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libitertools-a295c739cb0c5115.rlib" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libitertools-a295c739cb0c5115.rmeta" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-26458fd927156f14.rlib" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-26458fd927156f14.rmeta" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libquote-36054bcec486f3c0.rlib" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libquote-36054bcec486f3c0.rmeta" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libregex-4e61c4979f33639c.rlib" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libregex-4e61c4979f33639c.rmeta" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libserde-b9f428a504ba38d4.rlib" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libserde-b9f428a504ba38d4.rmeta" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libsyn-faa18c63fc12a335.rlib" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps/libsyn-faa18c63fc12a335.rmeta" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libtokio-ee648a3a992ea021.rlib" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libtokio-ee648a3a992ea021.rmeta" "--extern" "clippy_test_deps=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/libclippy_test_deps.rlib" "--extern" "clippy_test_deps=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps/libclippy_test_deps-728ed027f0623fde.rmeta" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/quote-fc00b89c3984f431" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/pulldown-cmark-5d2703cd75133d86" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/tokio-988b1ca3d2062fb8" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/serde_json-06fa4ab4a6a2c54b" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/memchr-300f0ecb3b085bd3" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/libc-467a9bf5fecaa5b9" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/serde-b5ab3846bee6426e" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/camino-95d981076bbda069" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-task-14f797a867cbbbea" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/thiserror-60234973db2aebdd" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/syn-c0e5e037aa06912e" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-core-67d40d21860bf7e2" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/x86_64-unknown-linux-gnu/debug/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/proc-macro2-c56f6977d10d30ea" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-channel-18c61a4c728f3f63" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/parking_lot_core-ff12b1c9a236dc80" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/slab-084e9e34eeb9970c" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/lock_api-172e9730276f85f1" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/unicase-dc66fd2ee33a4c53" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/futures-util-d30abe96b847bc06" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/debug/build/semver-367f0564b96a3f38" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/tests/ui/needless_arbitrary_self_type_unfixable" "tests/ui/auxiliary/proc_macro_attr.rs" "--edition" "2021" "--crate-type" "proc-macro" "--emit=link"



Aux build from tests/ui/needless_arbitrary_self_type_unfixable.rs:1 failed
compilation of aux build failed failed with exit status: 1


full stderr:
error[E0308]: mismatched types
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:33:32
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
LL | |                 });
   | |                  ^
   | |                  |
   | |__________________expected `Span`, found `DelimSpan`
   |                    expected due to this
   |
   = note: this error originates in the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::Signature: quote::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:33:32
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
LL | |                 });
   | |                  ^
   | |                  |
   | |__________________the trait `quote::ToTokens` is not implemented for `syn::Signature`
   |                    required by a bound introduced by this call
   |
   = help: the following other types implement trait `quote::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 23 others
   = note: required for `&syn::Signature` to implement `quote::ToTokens`
   = note: this error originates in the macro `$crate::quote_token_with_context_spanned` which comes from the expansion of the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::Block: quote::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:33:32
   |
LL |                   let my_block = quote_spanned!( brace.span => {
   |  ________________________________^
LL | |                     // Should not trigger `empty_line_after_outer_attr`
LL | |                     #[crate_type = "lib"]
LL | |                     #sig #block
LL | |                     Vec::new()
LL | |                 });
   | |                  ^
   | |                  |
   | |__________________the trait `quote::ToTokens` is not implemented for `syn::Block`
   |                    required by a bound introduced by this call
   |
   = help: the following other types implement trait `quote::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 23 others
   = note: required for `&mut syn::Block` to implement `quote::ToTokens`
   = note: this error originates in the macro `$crate::quote_token_with_context_spanned` which comes from the expansion of the macro `quote_spanned` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `quote::__private::TokenStream: syn::__private::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:39:26
   |
LL |                 *block = parse_quote!(#my_block);
   |                          ^^^^^^^^^^^^^^^^^^^^^^^
   |                          |
   |                          the trait `syn::__private::ToTokens` is not implemented for `quote::__private::TokenStream`
   |                          required by a bound introduced by this call
   |
   = help: the following other types implement trait `syn::__private::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 298 others
   = note: this error originates in the macro `$crate::__private::quote::quote` which comes from the expansion of the macro `parse_quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::ItemTrait: quote::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:43:23
   |
LL |     TokenStream::from(quote!(#item))
   |                       ^^^^^^^^^^^^^
   |                       |
   |                       the trait `quote::ToTokens` is not implemented for `syn::ItemTrait`
   |                       required by a bound introduced by this call
   |
   = help: the following other types implement trait `quote::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 23 others
   = note: this error originates in the macro `quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `syn::ItemImpl: quote::ToTokens` is not satisfied
Error:   --> tests/ui/auxiliary/proc_macro_attr.rs:96:23
   |
LL |     TokenStream::from(quote!(#item))
   |                       ^^^^^^^^^^^^^
   |                       |
   |                       the trait `quote::ToTokens` is not implemented for `syn::ItemImpl`
   |                       required by a bound introduced by this call
   |
   = help: the following other types implement trait `quote::ToTokens`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and 23 others
   = note: this error originates in the macro `quote` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 6 previous errors

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.


FAILURES:
    tests/ui/empty_line_after_doc_comments.rs
    tests/ui/empty_line_after_outer_attribute.rs
    tests/ui/needless_arbitrary_self_type_unfixable.rs

test result: FAIL. 3 tests failed, 867 tests passed, 4 ignored, 0 filtered out
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: tests failed
```

</details>

Cc @rust-lang/clippy

--- a/src/tools/clippy/tests/ui/empty_line_after_doc_comments.rs
+++ b/src/tools/clippy/tests/ui/empty_line_after_doc_comments.rs
@@ -1,4 +1,7 @@
 //@aux-build:proc_macro_attr.rs:proc-macro
+// Flaky test, see https://github.com/rust-lang/rust/issues/113585.
+//@ignore-32bit
+//@ignore-64bit
 #![warn(clippy::empty_line_after_doc_comments)]
 #![allow(clippy::assertions_on_constants)]
 #![feature(custom_inner_attributes)]

--- a/src/tools/clippy/tests/ui/empty_line_after_outer_attribute.rs
+++ b/src/tools/clippy/tests/ui/empty_line_after_outer_attribute.rs
@@ -1,4 +1,7 @@
 //@aux-build:proc_macro_attr.rs:proc-macro
+// Flaky test, see https://github.com/rust-lang/rust/issues/113585.
+//@ignore-32bit
+//@ignore-64bit
 #![warn(clippy::empty_line_after_outer_attr)]
 #![allow(clippy::assertions_on_constants)]
 #![feature(custom_inner_attributes)]

--- a/src/tools/clippy/tests/ui/needless_arbitrary_self_type_unfixable.rs
+++ b/src/tools/clippy/tests/ui/needless_arbitrary_self_type_unfixable.rs
@@ -1,4 +1,7 @@
 //@aux-build:proc_macro_attr.rs:proc-macro
+// Flaky test, see https://github.com/rust-lang/rust/issues/113585.
+//@ignore-32bit
+//@ignore-64bit
 
 #![warn(clippy::needless_arbitrary_self_type)]
 

